### PR TITLE
Add NotLoggedIfNull attribute

### PR DESCRIPTION
**Pull request by avtc**
Add `NotLoggedIfNullAttribute` and tests

It has effect only on reference types and nullables.

Ignore null properties can be globally applied without applying attributes:
```csharp
var log = new LoggerConfiguration()
  .Destructure.UsingAttributes(x => x.IgnoreNullProperties = true)
```

Based on PR: https://github.com/destructurama/attributed/pull/38

### CODE REVIEW

File: src/Destructurama.Attributed/Attributed/IPropertyDestructuringAttribute.cs [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+using System;
```
Comment (sungam3r) [inline src/Destructurama.Attributed/Attributed/IPropertyDestructuringAttribute.cs:15]: ?

File: src/Destructurama.Attributed/Attributed/LogMaskedAttribute.cs [hunk 2]
```diff
@@ -40,7 +41,14 @@ internal bool IsDefaultMask()
 
         internal object FormatMaskedValue(object propValue)
         {
-            var val = propValue as string;
+            if (propValue == null)
+                return null;
+
+            string val;
+            if (propValue is string)
+                val = propValue as string;
```
Comment (sungam3r) [inline src/Destructurama.Attributed/Attributed/LogMaskedAttribute.cs:49]:
```suggestion
            if (propValue is string s)
                val = s;
```

File: src/Destructurama.Attributed/Attributed/NotLoggedAttribute.cs [hunk 3]
```diff
@@ -13,21 +13,16 @@
...
-using Serilog.Events;
 
 namespace Destructurama.Attributed
 {
     /// <summary>
     /// Specified that a property should not be included when destructuring an object for logging.
     /// </summary>
     [AttributeUsage(AttributeTargets.Property)]
-    public class NotLoggedAttribute : Attribute, IPropertyDestructuringAttribute
+    public class NotLoggedAttribute : Attribute, IPropertyOptionalIgnoreAttribute
```
Comment (sungam3r) [inline src/Destructurama.Attributed/Attributed/NotLoggedAttribute.cs:23]: breaking change

Comment (avtc) [reply src/Destructurama.Attributed/Attributed/NotLoggedAttribute.cs]: Reverted

**Comment by avtc** [comment]
Related to #18

**Comment by sfektsz** [comment]
Any update on this, please?

**Comment by sungam3r** [comment]
I'm neither an owner of this repo nor have write access to be able to merge PRs.

**Comment by heman-dookhee-cko** [comment]
@SimonCropp Is it possible to have an update on this, please?

**Comment by sungam3r** [comment]
@heman-dookhee-cko Thanks for PR. Now I'm a member of destructurama org so please resolve conflicts and I'll review/merge.

**Comment by avtc** [comment]
@sungam3r conflicts resolved, please review/merge

File: src/Destructurama.Attributed/Attributed/LogMaskedAttribute.cs [hunk 4]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 using System;
+using System.Globalization;
```
Comment (sungam3r) [inline src/Destructurama.Attributed/Attributed/LogMaskedAttribute.cs:16]:
```suggestion
```

File: src/Destructurama.Attributed/Attributed/AttributedDestructuringPolicyOptions.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+namespace Destructurama.Attributed
+{
+    /// <summary>
+    /// Global destucturing options
```
Comment (sungam3r) [inline src/Destructurama.Attributed/Attributed/AttributedDestructuringPolicyOptions.cs:4]:
```suggestion
    /// Global destructuring options.
```

File: test/Destructurama.Attributed.Tests/MaskedAttributeTests.cs [hunk 6]
```diff
@@ -890,5 +891,6 @@ public void LogMaskedAttribute_Shows_First_NChars_And_Last_NChars_Then_Replaces_
             Assert.IsTrue(props.ContainsKey("ShowFirstAndLastThreeAndCustomMaskInTheMiddlePreservedLengthIgnored"));
             Assert.AreEqual("123_REMOVED_321", props["ShowFirstAndLastThreeAndCustomMaskInTheMiddlePreservedLengthIgnored"].LiteralValue());
         }
+
```
Comment (sungam3r) [inline test/Destructurama.Attributed.Tests/MaskedAttributeTests.cs:894]:
```suggestion
```

**Comment by sungam3r** [comment]
OK, will review soon. I was busy updating other 2 repos in destructurama org.

File: src/Destructurama.Attributed/Attributed/AttributedDestructuringPolicyOptions.cs [hunk 7]
```diff
@@ -0,0 +1,15 @@
+namespace Destructurama.Attributed
+{
+    /// <summary>
+    /// Global destructuring options.
+    /// </summary>
+    public class AttributedDestructuringPolicyOptions
+    {
+        /// <summary>
+        /// By setting IgnoreNullProperties to true no need to set [NotLoggedIfNull] for every logged property.
+        /// Custom types implemenenting IEnumerable, will be destructed as StructureValue and affected by IgnoreNullProperties
```
Comment (sungam3r) [inline src/Destructurama.Attributed/Attributed/AttributedDestructuringPolicyOptions.cs:10]:
```suggestion
        /// Custom types implementing IEnumerable, will be destructed as StructureValue and affected by IgnoreNullProperties 
```

File: README.md [hunk 8]
```diff
@@ -69,6 +69,59 @@ log.Information("Logging in {@Command}", command);
 <sup><a href='/test/Destructurama.Attributed.Tests/Snippets.cs#L44-L47' title='Snippet source file'>snippet source</a> | <a href='#snippet-logcommand' title='Start of snippet'>anchor</a></sup>
 <!-- endSnippet -->
 
+#### Ignoring a property if it's the default value
```
Comment (sungam3r) [inline README.md:72]:
```suggestion
#### Ignoring a property if it has the default value
```

**Comment by sungam3r** [comment]
@avtc Thanks. Will be published coming days.

**Comment by PawelTroka** [comment]
@sungam3r can this be causing a memory leak?
we have observed a huge memory usage increase after using this

**Comment by sungam3r** [comment]
Please open issue if so.

**Pull request merged**